plugins/qtosgrave/qtosgviewer.cpp

```diff
@@ -2145,14 +2157,16 @@ void QtOSGViewer::_PostToGUIThread(const boost::function<void()>& fn, GUIThreadQ
     boost::mutex::scoped_lock lockmsg(_mutexGUIFunctions);
     GUIThreadFunctionPtr pfn(new GUIThreadFunction(fn, block));
     // Block non-essential fucntions if viewer is not processing any messages
-    if (!_bUpdateEnvironment && priority < GUIThreadQueuePriority::HIGH) {
-        RAVELOG_WARN("GUI thread function of priority %d dropped due to _bUpdateEnvironment being set to false", priority);
+    if (!_bUpdateEnvironment && priority < ViewerCommandPriority::HIGH) {
+        RAVELOG_WARN("New GUI thread function of priority %d dropped due to _bUpdateEnvironment being set to false", priority);
         return;
     }
     // GUI thread function queue size limit
-    if (_GetQueueSizeForPriority(priority) >= _mapGUIFunctionListLimits[priority]) {
-        RAVELOG_WARN("GUI thread function of priority %d dropped due to function queue for given priority exceeding size of %d", priority, _mapGUIFunctionListLimits[priority]);
-        return;
+    if (_GetViewerCommandNumber(priority) >= _mapGUIFunctionListLimits[priority]) {
```

Comment:
Why we are comparing command number of several priority combined and a specific limit of one priority?

Reply:
We get the command number using all priorities at or above the input priority as a way to ensure that lower priority commands will always have equal or greater command numbers than higher priority commands and are therefore will either force an old command to be dropped or be dropped themselves at the same point as or an earlier point than higher priority commands.